crates/sui-types/src/move_package.rs

```diff
@@ -486,6 +486,19 @@ impl MovePackage {
         )
     }
 
+    pub fn deserialize_all_modules(&self) -> SuiResult<Vec<CompiledModule>> {
```

Comment:
I'm hoping this API would become unnecessary following the suggestion I made above about storing the snapshots in this form already.  This is a good thing, because we should in general avoid APIs that perform serialization/deserialization without being guarded by a max_binary_format_version like the API above.

It's not unsafe to use it the way you're using it now, it's just that having the API available on this type makes me nervous that it will one day be used in a place where we should be doing a version-bounded read.